python/shotgun_model/shotgun_model.py

```diff
@@ -696,6 +696,20 @@ def _update_item(self, item, data_item):
 
         self._set_tooltip(item, data_item.shotgun_data)
 
+    def _update_item_columns(self, item, data_item):
+        """
+        Updates the text content of the additional columns in the model item.
+        """
+        row = item.index().row()
+        if row >= 0:
+            for col_idx, column in enumerate(self.__column_fields, start=1):
+                sibling = self.itemFromIndex(self.index(row, col_idx))
+                column_text = sanitize_for_qt_model(data_item.shotgun_data.get(column))
+
+                if sibling and sibling != item and sibling.text() != column_text:
+                    self._log_debug(f"Updating sibling text content {sibling}")
+                    sibling.setText(column_text)
+
```

Comment:
Minor comment: I prefer fail-first strategies. I find it easier to read.


```suggestion
        row = item.index().row()
        if not row:
            return

        for col_idx, column in enumerate(self.__column_fields, start=1):
            sibling = self.itemFromIndex(self.index(row, col_idx))
            if not sibling or sibling != item:
                continue

            column_text = sanitize_for_qt_model(data_item.shotgun_data.get(column))         
            if sibling.text() == column_text:
                continue

            self._log_debug(f"Updating sibling text content {sibling}")
            sibling.setText(column_text)
```